pinot-plugins/pinot-metrics/pinot-dropwizard/src/main/java/org/apache/pinot/plugin/metrics/dropwizard/DropwizardGauge.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.plugin.metrics.dropwizard;
+
+import com.codahale.metrics.Gauge;
+import java.util.function.Function;
+import org.apache.pinot.spi.metrics.PinotGauge;
+
+
+public class DropwizardGauge<T> implements PinotGauge<T> {
+
+  private final Gauge<T> _gauge;
+
+  public DropwizardGauge(Gauge<T> gauge) {
+    _gauge = gauge;
+  }
+
+  public DropwizardGauge(Function<Void, T> condition) {
+    this(() -> condition.apply(null));
```

Comment:
is `this(...)` calling itself here?

Reply:
the output of this function is `T`

Reply:
oh, ic. Gauge<T> is a functional interface.
```
  public interface Gauge<T> extends Metric {
    T getValue();
  }
```